ddp-workspace/projects/ddp-sdk/src/lib/components/languageSelector.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, Inject, Input, OnInit } from "@angular/core";
+import { StudyLanguage } from "../models/studyLanguage";
+import { ConfigurationService } from "../services/configuration.service";
+import { LanguageService} from "../services/languageService.service";
+import { LanguageServiceAgent } from "../services/serviceAgents/languageServiceAgent.service";
+import { isNullOrUndefined } from "util";
+
+@Component({
+  selector: 'ddp-language-selector',
+  template: `
+    <div [hidden]="!loaded">
+      <button class="SimpleButton" [ngClass]="{'SimpleButton--Scrolled': isScrolled}" *ngIf="currentLanguage !== null && currentLanguage !== undefined" [matMenuTriggerFor]="menu">
+        <svg class="ddp-globe" height="24px" width="24px">
+          <title id="title" [lang]="currentLanguage.languageCode" translate>Toolkit.Header.LanguageSelection</title>
+          <use [attr.href]="iconURL"></use>
+        </svg>
+        <span class="ddp-current-language">{{currentLanguage.displayName}}</span>
+        <mat-icon class="ddp-dropdown-arrow">arrow_drop_down</mat-icon>
+      </button>
+      
+      <mat-menu #menu="matMenu">
+        <button mat-menu-item *ngFor="let lang of getUnselectedLanguages()" (click)="changeLanguage(lang)">{{lang.displayName}}</button>
+      </mat-menu>
+    </div>
+  `
+})
+export class LanguageSelectorComponent implements OnInit {
+  @Input() isScrolled: boolean;
+  public loaded: boolean;
+  public currentLanguage: StudyLanguage;
+  private studyLanguages: Array<StudyLanguage>;
+  public iconURL: string;
+
+  constructor (
+    private serviceAgent: LanguageServiceAgent,
+    private language: LanguageService,
+    @Inject('ddp.config') private config: ConfigurationService
+  ) { }
+
+  public ngOnInit(): void {
+    this.iconURL = this.config.languageSelectorIconURL ? this.config.languageSelectorIconURL : "assets/images/globe.svg#Language-Selector-3";
+    this.serviceAgent.getConfiguredLanguages(this.config.studyGuid).subscribe(x => {
```

Comment:
Please, take care of Observable subscription one of the following methods:
1. take(1) - https://github.com/broadinstitute/ddp-angular/blob/DDP-4719_LanguageSelector/ddp-workspace/projects/ddp-osteo/src/app/components/workflow-progress/workflow-progress.component.ts#L46
2. Save subscription and unsubscribe on destroying life cycle hook - https://github.com/broadinstitute/ddp-angular/blob/DDP-4719_LanguageSelector/ddp-workspace/projects/basil-app/src/app/loginLanding.component.ts#L37